src/compiler/toitdoc_parser.cc

```diff
@@ -951,28 +955,37 @@ int ToitdocParser::peek() {
   auto text = _toitdoc_source->text();
   ASSERT(_index <= _toitdoc_source->size());
   int c = text[_index];
-  if (c == '\n') {
-    // Note that this branch always returns, and that it never returns '\n' (only ' ' or '\0').
+  if (is_newline(c)) {
+    // Note that this branch always returns, and that it never returns '\r' or
+    //   '\n' (only ' ' or '\0').
     // Callers thus don't need to worry about '\n', but can simply check for
     //   whitespace by looking for spaces.
     if (is_single_line) return '\0';
     if (_next_index != -1) {
       // We already computed the indentation once and know that we aren't at a dedent.
       return ' ';
     }
-    _next_index = _index + 1;
+    if (c == '\r' && text[_index + 1] == '\n') {
```

Comment:
I guess we are assuming null termination here and below
to avoid out of bounds read?

Reply:
yes. added a comment.